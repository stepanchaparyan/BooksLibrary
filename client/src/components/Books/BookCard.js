import React, { Component } from 'react';
import PropTypes from 'prop-types';
import userImage from '../../assets/user.jpeg';
import { Link } from 'react-router-dom';
import { BookCardContainer,
         BookCardImage,
         BookCardBody,
         BookCardName,
         BookCardGenre
         } from './BookCardStyled';
import M from '../../Messages';

class BookCard extends Component {
    static propTypes = {
        book: PropTypes.any,
        onClick: PropTypes.any
    }

    render(){
        const { book } = this.props;
        return(
            <BookCardContainer>
                <BookCardImage src={userImage} alt="image"/>
                <BookCardBody className="book-card__body">
                    <BookCardName onClick={ this.props.onClick }>{M.get('books.name')}: {book.name}</BookCardName>
                    <BookCardGenre>{M.get('books.genre')}: {book.genre}</BookCardGenre>
                    <Link to={'/book/' + book.id}>{M.get('books.authorDataHere')}</Link>
                </BookCardBody>
            </BookCardContainer>
        );
    }
}

export { BookCard };