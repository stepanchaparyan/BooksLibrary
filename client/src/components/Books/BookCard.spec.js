import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BookCard } from "./BookCard.js";

const props = {
  book: {name: 'Book1', genre: 'Drama', id: '123'},
  onClick: () => {},
}

function setupShallow() {
  const wrapper = shallow(<BookCard {...props} />);
  return { wrapper };
}

describe('BookCard component test with Enzyme', () => {
  it('renders without crashing', () => {
    const { wrapper } = setupShallow();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});


