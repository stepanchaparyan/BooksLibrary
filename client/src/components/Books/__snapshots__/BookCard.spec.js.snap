// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard component test with Enzyme renders without crashing 1`] = `
<styled.div>
  <styled.img
    alt="image"
    src="https://cdn.iconscout.com/icon/free/png-256/avatar-372-456322.png"
  />
  <styled.div
    className="book-card__body"
  >
    <styled.div
      onClick={[Function]}
    >
      Name
      : 
      Book1
    </styled.div>
    <styled.div>
      Genre
      : 
      Drama
    </styled.div>
    <Link
      to="/book/123"
    >
      Author data here
    </Link>
  </styled.div>
</styled.div>
`;
